src/search_internal.h

```diff
@@ -125,7 +125,7 @@ struct SearchIterator::InternalData {
     }
 
     Entry& get_entry() {
-        if ( !_entry ) {
+        if (!_entry || !document_fetched) {
```

Comment:
Why adding this test ?
Can we have a entry but no document_fetched ? (knowing that we build the entry from the document)

Reply:
Aah my bad, I must have added it in some testing and forgot to remove it. I'll fix it back quickly.